data/src/main/java/com/linkedin/data/schema/annotation/AnnotationCheckResolvedPropertiesVisitor.java

```diff
@@ -81,7 +71,7 @@ public void callbackOnContext(TraverserContext context, DataSchemaTraverse.Order
               chooseProperties(schemaField.getResolvedProperties(), schemaField.getProperties())));
       pathToSchema.removeLast();
     }
-    else if (unionMember!= null && pathToSchema.getLast().equals(context.getEnclosingUnionMember().getUnionMemberKey()))
+    else if (unionMember!= null && pathToSchema.peekLast().equals(context.getEnclosingUnionMember().getUnionMemberKey()))
```

Comment:
In theory it can, which is why the IDE warns. Practically it cannot here because if the enclosing field or union member was present then we are guaranteed to have elements in the queue.